boilerplate/app/components/screen/screen.tsx

```diff
@@ -32,6 +32,16 @@ function ScreenWithScrolling(props: ScreenProps) {
   const backgroundStyle = props.backgroundColor ? { backgroundColor: props.backgroundColor } : {}
   const insetStyle = { paddingTop: props.unsafe ? 0 : insets.top }
 
+  // The followings for <Screen preset='auto'/>
+  // This will automatically disables scrolling if content fits the screen.
+  const { height } = Dimensions.get('window');
+  const [screenHeight, setScreenHeight] = React.useState(0);
+  const scrollEnabled = props.preset === 'scroll' || props.preset === 'auto' && screenHeight > height * presets.auto.offset.percent - presets.auto.offset.point;
+
+  const onContentSizeChange = (width, height) => {
+    setScreenHeight(height);
```

Comment:
Will this cause a double-render?

In other words, does `onContentSizeChange` come with a re-render anyway? If so, we could just hold the screenheight in a useRef and calculate when the re-render happens.

It's possible this simply gets batched with the onContentSizeChange render too. Would need to test this, I guess!

Reply:
Thank you for your response!

Yeah, you are right about double-rendering.

I've made some changes and pushed a new commit. Can you check?

With this fix, it's not rendering twice by default, only if preset '**auto**' is used and content fits the screen, so it's necessary.

There is one setback though, metro's soft reload is not updating **scrollEnabled** state properly when constantly changing between preset '**auto**' and '**scroll**' because **onContentSizeChange** not called either on a file change (at least if content stays the same). One may need reload screen manually to catch the updated state.

It's something I can live with. To fix this issue, it needs significantly more code which probably makes code readability worse.

Reply:
> There is one setback though, metro's soft reload is not updating scrollEnabled state properly when constantly changing between preset 'auto' and 'scroll' because onContentSizeChange not called either on a file change (at least if content stays the same). One may need reload screen manually to catch the updated state.

I pushed one more commit to fix this and works well but it's more code as I said.